script.speedfaninfo/default.py

```diff
@@ -0,0 +1,330 @@
+import datetime, os
+from kodi_six import xbmc, xbmcaddon, xbmcgui, xbmcvfs
+from threading import Thread
+try:
+    from itertools import izip_longest as _zip_longest
+except ImportError:
+    from itertools import zip_longest as _zip_longest
+from resources.common.xlogger import Logger
+from resources.common.kodisettings import getSettingBool, getSettingInt, getSettingString
+
+### get addon info and set globals
+addon        = xbmcaddon.Addon()
+addonname    = addon.getAddonInfo( 'id' )
+addonversion = addon.getAddonInfo( 'version' )
+addonpath    = addon.getAddonInfo( 'path' )
+addonicon    = xbmc.translatePath( '%s/icon.png' % addonpath )
+language     = addon.getLocalizedString
+preamble     = '[SpeedFan Info]'
+logdebug     = addon.getSetting( 'logging' )
+
+lw = Logger( preamble = preamble, logdebug = logdebug )
+
+#global used to tell the worker thread the status of the window
+windowopen   = True
+
+#capture a couple of actions to close the window
+ACTION_PREVIOUS_MENU = 10
+ACTION_BACK = 92
+
+
+def updateWindow( name, w ):
+    delay = getSettingInt( addon, 'update_delay', default=30 )
+    while windowopen and (not xbmc.Monitor().abortRequested):
```

Comment:
`xbmc.Monitor().abortRequested()`
You need to call into the method otherwise you receive a func object